benchmarks/perf-tool/okpt/io/config/parsers/util.py

```diff
@@ -6,30 +6,24 @@
 
 """Utility functions for parsing"""
 
-from dataclasses import dataclass
-from typing import Union, cast
-import h5py
 
 from okpt.io.config.parsers.base import ConfigurationError
+from okpt.io.dataset import HDF5DataSet, BigANNNeighborDataSet, \
+    BigANNVectorDataSet, DataSet, Context
 
 
-@dataclass
-class Dataset:
-    train: h5py.Dataset
-    test: h5py.Dataset
-    neighbors: h5py.Dataset
-    distances: h5py.Dataset
+def parse_dataset(dataset_format: str, dataset_path: str,
+                  context: Context) -> DataSet:
+    if dataset_format == 'hdf5':
+        return HDF5DataSet(dataset_path, context)
 
+    if dataset_format == 'bigann' and context == Context.NEIGHBORS:
+        return BigANNNeighborDataSet(dataset_path)
```

Comment:
nit: BigANNNeighborDataset

Reply:
I felt like that looked better as well but PyCharm was giving me spelling errors so I went with DataSet. Here is a post on it: https://english.stackexchange.com/questions/2120/which-is-correct-dataset-or-data-set. I dont really have a preference so I went with the one that wasnt giving me spelling errors.